test_scripts/API/Restructuring_OnResetTimeout/common_OnResetTimeout.lua

```diff
@@ -0,0 +1,463 @@
+  c.getHMIConnection():ExpectRequest("TTS.Speak",
+    { ttsChunks = paramsAlert.ttsChunks,
+    speakType = "ALERT",
+    appID = c.getHMIAppId()
+  })
+  :Do(function(_, data)
+    c.getHMIConnection():SendNotification("TTS.Started")
+    local function SpeakResponse()
+      c.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { })
+      c.getHMIConnection():SendNotification("TTS.Stopped")
+    end
+    RUN_AFTER(SpeakResponse, 2000)
+  end)
+  c.getMobileSession():ExpectResponse(cid, pRespParams)
+  :Timeout(pRespExpect)
+  :ValidIf(function()
+    local respTime = timestamp()
+    local timeBetweenRespAndNot = respTime - c.notificationTime
+    if timeBetweenRespAndNot >= pRespTime - 500 and timeBetweenRespAndNot <= pRespTime + 500 then
+      return true
+    else
+      return false, "Response is received in some unexpected time. Actual time is " .. timeBetweenRespAndNot ..
+      ". Ecpected time is " .. pRespTime
+    end
+  end)
+end
+
+function c.createInteractionChoiceSet()
+  local params = {
+    interactionChoiceSetID = 100,
+    choiceSet = {
+      {
+        choiceID = 111,
+        menuName = "Choice111",
+        vrCommands = { "Choice111" }
+      }
+    }
+  }
+  local corId = c.getMobileSession():SendRPC("CreateInteractionChoiceSet", params)
+  c.getHMIConnection():ExpectRequest("VR.AddCommand")
+  :Do(function(_, data)
+     c.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+  c.getMobileSession():ExpectResponse(corId, { success = true, resultCode = "SUCCESS" })
+end
+
+function c.PerformInteraction( pRespExpect, pRespTime, pResponseFunc, pParams, pRespParams )
+  if not pRespParams then pRespParams = { success = true, resultCode = "SUCCESS" } end
+  local params = {
+    initialText = "StartPerformInteraction",
+    interactionMode = "VR_ONLY",
+    interactionChoiceSetIDList = { 100 },
+    initialPrompt = {
+      { type = "TEXT", text = "pathToFile1" }
+    }
+  }
+  local corId = c.getMobileSession():SendRPC("PerformInteraction", params)
+  c.getHMIConnection():ExpectRequest("UI.PerformInteraction")
+  :Do(function(_, data)
+    c.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+  end)
+  c.getHMIConnection():ExpectRequest("VR.PerformInteraction", {
+    initialPrompt = params.initialPrompt
+  })
+  :Do(function(_, data)
+    pResponseFunc(data, pParams)
+  end)
+  c.getMobileSession():ExpectResponse(corId, pRespParams)
+  :Timeout(pRespExpect)
+  :ValidIf(function()
+    local respTime = timestamp()
+    local timeBetweenRespAndNot = respTime - c.notificationTime
+    if timeBetweenRespAndNot >= pRespTime - 500 and timeBetweenRespAndNot <= pRespTime + 500 then
+      return true
+    else
+      return false, "Response is received in some unexpected time. Actual time is " .. timeBetweenRespAndNot ..
+      ". Ecpected time is " .. pRespTime
+    end
+  end)
+end
+
+function c.Slider( pRespExpect, pRespTime, pResponseFunc, pParams, pRespParams )
```

Comment:
@HSavynetska @GetmanetsIrina please notice that _Slider_ request has a mandatory param "_timeout_" with _default_ _value_ = 10000. https://github.com/smartdevicelink/sdl_core/blob/develop/src/components/interfaces/MOBILE_API.xml#L5700 
So final  timeout for Slider request will be = default timeout + internal timeout (from message) = 10000 + 10000 = 20000